samza-kv/src/main/scala/org/apache/samza/storage/kv/SerializedKeyValueStore.scala

```diff
@@ -59,23 +59,23 @@ class SerializedKeyValueStore[K, V](
     val valBytes = toBytesOrNull(value, msgSerde)
     store.put(keyBytes, valBytes)
     metrics.puts.inc
-    metrics.maxRecordSizeBytes.set(Math.max(metrics.maxRecordSizeBytes.getValue, valBytes.length))
+    while({val max = metrics.maxRecordSizeBytes.getValue; !metrics.maxRecordSizeBytes.compareAndSet(max, Math.max(max, valBytes.length))}){}
```

Comment:
Although this format is compact, IMO the syntax looks a little weird. Maybe just write a more basic loop that looks more like Java syntax. Also, could you pull this into a helper method to share with `putAll`?

Reply:
sounds good